cast/src/starknet_commands/declare.rs

```diff
@@ -14,36 +13,64 @@ use starknet::{
     signers::LocalWallet,
 };
 use std::sync::Arc;
+use std::process::Command;
 
 #[derive(Args)]
 #[command(about = "Declare a contract to starknet", long_about = None)]
 pub struct Declare {
-    /// Path to the sierra compiled contract
-    pub sierra_contract_path: Utf8PathBuf,
-
-    /// Path to the casm compiled contract
-    pub casm_contract_path: Utf8PathBuf,
+    /// contract name
+    pub contract: String,
 
     /// Max fee for the transaction. If not provided, max fee will be automatically estimated
     #[clap(short, long)]
     pub max_fee: Option<u128>,
 }
 
 pub async fn declare(
-    sierra_contract_path: &Utf8PathBuf,
-    casm_contract_path: &Utf8PathBuf,
+    contract: &str,
     max_fee: Option<u128>,
     account: &mut SingleOwnerAccount<&JsonRpcClient<HttpTransport>, LocalWallet>,
 ) -> Result<DeclareTransactionResult> {
+    let _ = Command::new("scarb")
+        .current_dir(std::env::current_dir().expect("failed to obtain current dir"))
+        .arg("build")
+        .output()
+        .expect("Failed to build contracts with Scarb");
```

Comment:
how does this work, it builds all the contracts in current directory?

Reply:
yes

Reply:
btw, is it possible to compile only one contract (the one we want to declare) ?

Reply:
> btw, is it possible to compile only one contract (the one we want to declare) ?
I don't think so, but let's ask @maciektr

Reply:
@THenry14 @piotmag769 
It builds all contracts in Scarb project indicated by current directory. 
As of now, it is not possible to indicate which contracts to build.

Reply:
cool, thanks for clarification.
in that case, does it make sense to also expect an error, where we can't find Scarb.toml in current or parent directories?

Reply:
Scarb will raise an error if there's no `Scarb.toml` I think